metabase/metabase
# Fix wrongly used `Database` and `Table` types from `metabase-types/api`

Epic https://github.com/metabase/metabase/issues/30763

`Database` and `Table` entity loaders always return `metabase-lib` entities (because they have `getObject` selector overridden). This PR fixes where we use `Database` and `Table` types from `metabase-types/api` and uses ones from `metabase-lib/metadata`. No app code changes, only types + tests.

How to verify:
- type check, unit tests are green on CI


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30661)
